swatch-core/src/main/java/org/candlepin/subscriptions/registry/TagProfile.java

```diff
@@ -243,4 +248,70 @@ public String tagForOfferingProductName(String offeringProductName) {
   public Set<String> getOfferingProductNamesForTag(String productTag) {
     return tagToOfferingProductNamesLookup.getOrDefault(productTag, Collections.emptySet());
   }
+  // ProductRegistry methods below
+  /**
+   * Verify that the granularity requested is compatible with the finest granularity supported by
+   * the product. For example, if the requester asks for HOURLY granularity but the product only
+   * supports DAILY granularity, we can't meaningfully fulfill that request.
+   *
+   * @throws IllegalStateException if the granularities are not compatible
+   */
+  public void validateGranularityCompatibility(
+      ProductId productId, Granularity requestedGranularity) {
+    if (!tagSupportsGranularity(productId.toString(), requestedGranularity)) {
+      String msg =
+          String.format(
+              "%s does not support any granularity finer than %s",
+              productId, requestedGranularity.getValue());
+      throw new IllegalStateException(msg);
+    }
+  }
+
+  public Map<Integer, Set<String>> getEngProductIdToSwatchProductIdsMap() {
```

Comment:
Let's create this mapping when the object is initialized.  Take a look at `handleTagMapping` and build it there and allow access via a getter.

Reply:
Hey Micheal, I had this method so that I could return a map of <Integer,String> datatypes for Capacity processing, While the `tagToEngProductsLookup` is a <String, Set<String>> map. I changed the other methods based on your suggestions and added a new lookup for arch in the `handleTagMapping`, But I want to leave this methods as is. As creating this mapping in the lookup would be somewhat redundant as it would hold the same information as  `tagToEngProductsLookup` only that the keys are Integers.